ClassLibraryProjects/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Data;
+using System.Runtime.InteropServices;
+
+namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        private Node<T>? _Next;
+        public T Value { get; set; }
+        
+        // I don't know how to unit test for null with the way Next is being used. //
+        public Node<T> Next
+        {
+            get
+            {
+                return _Next!;
+            }
+            private set
+            { 
+                _Next = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+        public Node(T value) 
+        { 
+            Value = value;
+            Next = this;
+        }
+
+        public override string ToString() 
+        {
+            return this.Value!.ToString()!;
+        }
+
+        public void Append(Node<T> nextNode) 
+        { 
+            if (this.Exists(nextNode.Value))
+            {
+                throw new ArgumentException("Duplicate Value detected");
+            }
+
+            nextNode.Next = this.Next;
+            this.Next = nextNode;    
+        }
```

Comment:
```suggestion
        public void Append(T value) 
        { 
            if (Exists(value))
                throw new ArgumentException("Duplicate Value detected");
            Node<T> newNode = new(value);
            newNode.Next = Next;
            Next = newNode;    
        }
```
Append should take in a T value, not a Node of type T. Using the "this" keyword here is also not necessary. You could argue it improves readability, but the IDE recommends simplification. This will break a lot of tests. Just have the tests pass in the values instead of actual nodes.

Reply:
Fixed Append and the tested it would break